src/RelayedTransactionValidator.ts

```diff
@@ -0,0 +1,95 @@
+import log from 'loglevel';
+import type { Transaction } from 'ethers';
+import type { ContractInteractor } from '@rsksmart/rif-relay-common';
+import type {
+  EnvelopingConfig,
+  RelayTransactionRequest,
+  DeployTransactionRequest,
+} from '@rsksmart/rif-relay-common/dist/src';
+
+export class RelayedTransactionValidator {
+  private readonly contractInteractor: ContractInteractor;
+
+  private readonly config: EnvelopingConfig;
+
+  constructor(
+    contractInteractor: ContractInteractor,
+    config: EnvelopingConfig
+  ) {
+    this.contractInteractor = contractInteractor;
+    this.config = config;
+  }
+
+  /**
+   * Decode the signed transaction returned from the Relay Server, compare it to the
+   * requested transaction and validate its signature.
+   */
+  validateRelayResponse(
+    request: RelayTransactionRequest | DeployTransactionRequest,
+    { v, r, s, to, data, value, from, nonce }: Transaction,
+    relayWorker: string
+  ): void {
+    log.info('validateRelayResponse - Transaction is', {
+      v,
+      r,
+      s,
+      to,
+      from,
+      data,
+      value,
+      nonce,
+    });
+
+    if (!to) {
+      throw Error('Transaction has no recipient address');
+    }
+
+    if (!from) {
+      throw Error('Transaction has no signer');
+    }
+
+    const isDeploy = (request as DeployTransactionRequest).relayRequest.request
+      .recoverer
+      ? true
+      : false;
```

Comment:
we should probably create a utility function for this, since we often check if a request is a deploy or no. (e.g.: a similar code can be found in the AccountManager also)

Reply:
this is done